### PR TITLE
API tutorial

### PR DESCRIPTION
**Pull request by bradleycamacho**

### CODE REVIEW

File: src/content/docs/tutorial-external-services/respond-external.mdx [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+metaDescription: Learn how to manage APIs and other external services
+---
+
+import externalMaps from 'images/apm_screenshot-crop_external-maps.webp'
+
+import externalRPM from 'images/apm_screenshot-crop_maps-rpm.webp'
+
+import externalOverview from 'images/apm_screenshot-crop_maps-home.png'
+
+Almost all applications and services make calls to APIs and other external services. When something as simple as pressing the `buy` button on a site can make dozens of API calls, issues are bound to arise. Problematic API calls can result in anything from a simple inventory mishap to a code-red situations where your site can't communicate with your payment processor.
```
Comment (homelessbirds) [inline src/content/docs/tutorial-external-services/respond-external.mdx:12]: Repetition of the word simple. Also, opening with the word almost feels a little passive. Can you rephrase this to make it more direct? That's nit-picky so feel free to disregard.

Comment (homelessbirds) [inline src/content/docs/tutorial-external-services/respond-external.mdx:12]: code-red situation not situations

File: src/content/docs/tutorial-external-services/respond-external.mdx [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+import externalRPM from 'images/apm_screenshot-crop_maps-rpm.webp'
+
+import externalOverview from 'images/apm_screenshot-crop_maps-home.png'
+
+Almost all applications and services make calls to APIs and other external services. When something as simple as pressing the `buy` button on a site can make dozens of API calls, issues are bound to arise. Problematic API calls can result in anything from a simple inventory mishap to a code-red situations where your site can't communicate with your payment processor. 
+
+<img
+title="Maps overview"
+alt="Home maps overview"
+src={externalOverview}
```
Comment (homelessbirds) [inline src/content/docs/tutorial-external-services/respond-external.mdx:17]: I would move the screenshot. Here it feels like it breaks up the flow and I've lost the train of thought.

File: src/content/docs/tutorial-external-services/respond-external.mdx [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+Once you've installed an agent, go to **[one.newrelic.com](https://one.newrelic.com/nr1-core?filters=(domain%3D'APM'ANDtype%3D'APPLICATION'))** and select your app. If you don't see much data just yet, step away for a while and let the agent gather real-time data as your application runs.
+    </Step>
+    <Step>
+
+### Additional setup
+
+Your specific configuration may require additional setup. Take a look at our [external service setup doc](/docs/apm/apm-ui-pages/monitoring/external-services/external-services-setup/) for more information.
+
+Once you've completed any additional steps from the doc above, go to **[one.newrelic.com](https://one.newrelic.com/nr1-core?filters=(domain%3D'APM'ANDtype%3D'APPLICATION'))** > select your app > *External services**. You should see data about your APIs and external services flow in.
```
Comment (homelessbirds) [inline src/content/docs/tutorial-external-services/respond-external.mdx:93]:
```suggestion
Once you've completed any additional steps from the doc above, go to **[one.newrelic.com](https://one.newrelic.com/nr1-core?filters=(domain%3D'APM'ANDtype%3D'APPLICATION'))** > select your app > **External services**. You should see data about your APIs and external services flow in.
```

File: src/content/docs/tutorial-external-services/respond-external.mdx [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+</Steps>
+
+## Identify problematic APIs 
+
+You now have the ability to track how your application interacts with APIs and other external services. Let's use that info to identify which APIs or external services are causing issues:
+
+<Steps>
+    <Step>
+### Think about the root cause 
+
```
Comment (homelessbirds) [inline src/content/docs/tutorial-external-services/respond-external.mdx:104]: This whole doc is assuming that a user is coming to New Relic because they're getting some sort of feedback that their site/application isn't working. Is it possible to identify ways New Relic might let them know something is wrong? Or is the premise basically: my site isn't working, let me check API calls? Also-- how does this play with alerts/APM? I feel like there are opportunities to integrate and weave different areas together?

File: src/content/docs/tutorial-external-services/respond-external.mdx [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+Your application might make calls to dozens or hundred of APIs. While you continue with the following steps, keep your root issue in mind. If you know you have an issue with purchases you should focus on purchase and transacton related APIs. If it's a log in issue, you might focus on calls to user databases or external authentication services.
+    </Step>
+    <Step>
+
+### Triage with maps 
+
+Let's say you run an webstore and have received emails from your users complaining that they had not been notified when their package arrived. You find this strange as they should have received both SMS and email notifications.
+
+You've instrumented your service that handles the entire delivery process. Go to the **External services** page and then click *Maps**.
```
Comment (homelessbirds) [inline src/content/docs/tutorial-external-services/respond-external.mdx:115]:
```suggestion
You've instrumented your service that handles the entire delivery process. Go to the **External services** page and then click **Maps**.
```

File: src/content/docs/tutorial-external-services/solve-external.mdx [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+    alt="The preview pane for an external service"
+    src={apmPreview}
+    />  
+
+
+This opens up a pane where you get an overview of the general state of the service. You already used some of these metrics to identify it as the problem, but it's useful to take a deeper look here. Keep an eye out for:
+
+* Any spikes in the charts. 
+* Obviously high response rate or throughput.
+* If response rate and throughput hit highs and lows around the same time.
```
Comment (homelessbirds) [inline src/content/docs/tutorial-external-services/solve-external.mdx:42]:
```suggestion
* Do response rate and throughput hit highs and lows around the same time?
```

Comment (homelessbirds) [reply src/content/docs/tutorial-external-services/solve-external.mdx:42]: Not sure if this question works with the phrase `Keep an eye out for:` but it feels like it flows better. It's up to you!

File: src/content/docs/tutorial-external-services/solve-external.mdx [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+
+If you identify an anomaly, use the time picker in the top right to expand the timerange you are looking at. This is useful to identify an anomaly as opposed to a pattern.
+
+In the screenshot above, nothing stands out except the throughput is a bit higher than you would have expected. Not anything to worry about immediately.
```
Comment (homelessbirds) [inline src/content/docs/tutorial-external-services/solve-external.mdx:46]:
```suggestion
You can see in the screenshot above that nothing stands out except the throughput is a bit higher than you would have expected. There's nothing to worry about immediately.
```

File: src/content/docs/tutorial-external-services/solve-external.mdx [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+
+Click in as deep as you find useful, but focus again on finding anomalies or errors in the chart. For example, you might follow the trace all the way up to the API call that decrypts a user's stored phone number. An error here could uncover the whole problem. If the decrypt call fails and never returns a valid phone number, your user would never receive a phone notification. 
+
+This would also explain why you saw no errors in the `Sms notification` charts. That API was working correctly, but an API further upstream was giving it data that didn't actually serve a purpose.
+
+    </Step>
+    <Step>
+### Use the data to identify logical errors
+
+Not every solution comes directly from errors or anomalies. Sometimes the root cause can be much more mundane.
```
Comment (homelessbirds) [inline src/content/docs/tutorial-external-services/solve-external.mdx:69]:
```suggestion
Not every solution comes directly from errors or anomalies.  The root cause can be much more mundane.
```

File: src/content/docs/tutorial-external-services/solve-external.mdx [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+    title="The trace pane for an external service"
+    alt="The trace pane for an external service"
+    src={apmTrace}
+    />  
+
+This displays various charts about traces involved with that service. In this case, there is a single trace called `Controller/Sinatra//purchase` which allows you to follow the data created from your user clicking the buy button all the way to the API call that sends an SMS notification. 
+
+Click in as deep as you find useful, but focus again on finding anomalies or errors in the chart. For example, you might follow the trace all the way up to the API call that decrypts a user's stored phone number. An error here could uncover the whole problem. If the decrypt call fails and never returns a valid phone number, your user would never receive a phone notification. 
+
+This would also explain why you saw no errors in the `Sms notification` charts. That API was working correctly, but an API further upstream was giving it data that didn't actually serve a purpose.
```
Comment (homelessbirds) [inline src/content/docs/tutorial-external-services/solve-external.mdx:63]: Is it worth explaining upstream vs. downstream? Also maybe I am missing something, but how do you know it's a problem upstream?

**Review by homelessbirds — APPROVED** [review]

File: src/content/docs/tutorial-external-services/respond-external.mdx (hunk 4 above)
Comment (bradleycamacho) [reply src/content/docs/tutorial-external-services/respond-external.mdx:104]: This is a solid point. The audience for this is basically "I'm 90% sure I have an API issue, let me learn how to use New Relic to troubleshoot that one specific slice of my system."

Our _root causes_ tutorial is a more general "my app is slow" let me use APM, alerts, SLOs, etc. to take a wider look at my system.

This tutorial is a thin slice on purpose
